src/Symfony/Component/Security/Core/Util/ClassUtils.php

```diff
@@ -54,12 +40,6 @@ private function __construct()
      */
     public static function getRealClass($object)
     {
-        $class = is_object($object) ? get_class($object) : $object;
-
-        if (false === $pos = strrpos($class, '\\'.self::MARKER.'\\')) {
-            return $class;
-        }
-
-        return substr($class, $pos + self::MARKER_LENGTH + 2);
+        return AclClassUtils::getRealClass($object);
```

Comment:
I can do the same check as here: https://github.com/symfony/security-acl/pull/3/files#diff-f95db4a0923d2b083e7c0d2d111c722eR45

If it doesn't exist, do the old logic. That way people who do use everything (which is probsbly 99% of the people), will get the proper checks.